common/ast/src/parser/rule/expr.rs

```diff
@@ -0,0 +1,672 @@
+        },
+        |number| Literal::Number(number.text.to_string()),
+    );
+    let boolean = alt((
+        value(Literal::Boolean(true), rule! { TRUE }),
+        value(Literal::Boolean(false), rule! { FALSE }),
+    ));
+    let null = value(Literal::Null, rule! { NULL });
+
+    rule!(
+        #string
+        | #number
+        | #boolean
+        | #null
+    )(i)
+}
+
+pub fn type_name<'a, Error>(i: Input<'a>) -> IResult<Input<'a>, TypeName, Error>
+where Error: ParseError<Input<'a>> {
+    let ty_char = map(
+        rule! { CHAR ~ ("(" ~ #literal_u64 ~ ")")? },
+        |(_, opt_args)| TypeName::Char(opt_args.map(|(_, length, _)| length)),
+    );
+    let ty_varchar = map(
+        rule! { VARCHAR ~ ("(" ~ #literal_u64 ~ ")")? },
+        |(_, opt_args)| TypeName::Varchar(opt_args.map(|(_, length, _)| length)),
+    );
+    let ty_float = map(
+        rule! { FLOAT ~ ("(" ~ #literal_u64 ~ ")")? },
+        |(_, opt_args)| TypeName::Float(opt_args.map(|(_, prec, _)| prec)),
+    );
+    let ty_int = map(
+        rule! { INTEGER ~ ("(" ~ #literal_u64 ~ ")")? },
+        |(_, opt_args)| TypeName::Int(opt_args.map(|(_, display, _)| display)),
+    );
+    let ty_tiny_int = map(
+        rule! { TINYINT ~ ("(" ~ #literal_u64 ~ ")")? },
+        |(_, opt_args)| TypeName::TinyInt(opt_args.map(|(_, display, _)| display)),
+    );
+    let ty_small_int = map(
+        rule! { SMALLINT ~ ("(" ~ #literal_u64 ~ ")")? },
+        |(_, opt_args)| TypeName::SmallInt(opt_args.map(|(_, display, _)| display)),
+    );
+    let ty_big_int = map(
+        rule! { BIGINT ~ ("(" ~ #literal_u64 ~ ")")? },
+        |(_, opt_args)| TypeName::BigInt(opt_args.map(|(_, display, _)| display)),
+    );
+    let ty_real = value(TypeName::Real, rule! { REAL });
+    let ty_double = value(TypeName::Double, rule! { DOUBLE });
+    let ty_boolean = value(TypeName::Boolean, rule! { BOOLEAN });
+    let ty_date = value(TypeName::Date, rule! { DATE });
+    let ty_time = value(TypeName::Time, rule! { TIME });
+    let ty_timestamp = value(TypeName::Timestamp, rule! { TIMESTAMP });
+    let ty_text = value(TypeName::Text, rule! { TEXT });
+
+    rule!(
+        #ty_char
+        | #ty_varchar
+        | #ty_float
+        | #ty_int
+        | #ty_tiny_int
+        | #ty_small_int
+        | #ty_big_int
+        | #ty_real
+        | #ty_double
+        | #ty_boolean
+        | #ty_date
+        | #ty_time
+        | #ty_timestamp
+        | #ty_text
+    )(i)
+}
+
+// TODO(andylokandy): complete the keyword-function list, or remove the functions' name from keywords
+pub fn function_name<'a, Error>(i: Input<'a>) -> IResult<Input<'a>, String, Error>
+where Error: ParseError<Input<'a>> {
+    map(
+        rule! {
+            Ident
+            | COUNT
+            | SUM
+            | AVG
+            | MIN
+            | MAX
+            | STDDEV_POP
+            | SQRT
+        },
+        |name| name.text.to_string(),
+    )(i)
```

Comment:
@leiysky FYI, this list is probably incomplete. Which solution mentioned in the todo do you prefer?

Reply:
I suppose we just keep `COUNT` as a token and use `Ident` for the rest, i.e. the second way?

Reply:
> I suppose we just keep `COUNT` as a token and use `Ident` for the rest, i.e. the second way?

Agree.